features/step_definitions/helper_services.rb

```diff
@@ -153,19 +153,48 @@
   step %Q/evaluation of `pod` is stored in the :proxy_pod clipboard/
 end
 
-Given /^I have LDAP service in my project$/ do
-    ###
-    # The original idea is trying to put ldap server in the openshift to make this flexy.
-    # Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
-    # 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
-    # 2, Port forward the ldap server pod to the jenkins slave.
-    # So take the second one since this one can be implemented currently
-    ###
+###
+# The original idea is trying to put ldap server in the openshift to make this flexy.
+# Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
+# 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
+# 2, Port forward the ldap server pod to the jenkins slave.
+# So take the second one since this one can be implemented currently
+###
+Given /^I have LDAP service in my project with clusterversion "([^"]*)"$/ do |clusterversion|
+  if clusterversion == '4.5' || '4.6'
```

Comment:
Please keep original step definition.

If necessary to check cluster version please use `env.version_ge("4.6", user: user)` instead of expecting user to set version.

But how about use the `run` command like before. Ignore cluster creating a RC or not. Just wait for pod by label. Did you try that?

Reply:
I have tried with `run` with RC and its giving me labels with `app: ldapserver` in 4.6 and `run: ldapserver` in 4.3 , so that was not working for me. Removed RC and its working.